src/JsonAsynchronousNodeKit/LargeJsonParser.cs

```
@@ -0,0 +1,281 @@
+
+        this.isFinalBlock = reader.IsFinalBlock;
+        this.readerState = reader.CurrentState;
+
+        return this.GetArray(objType);
+    }
+
+    private IEnumerable<object> GetArray(Type type)
+    {
+        this.enumeratorActive = true;
+        while (true)
+        {
+            var obj = this.ReadArrayObject(type);
+            if (obj is not null)
+            {
+                yield return obj;
+            }
+            else
+            {
+                this.enumeratorActive = false;
+                yield break;
+            }
+        }
+    }
+
+    private object? ReadArrayObject(Type type)
+    {
+        try
+        {
+            var reader = new Utf8JsonReader(this.buffer, this.isFinalBlock, this.readerState);
+
+            ParserUtils.Read(this.stream, ref this.buffer, ref reader);
+
+            object? result;
+            if (reader.TokenType == JsonTokenType.EndArray)
+            {
+                result = null;
+            }
+            else
+            {
+                result = this.GetObject(type, ref reader);
+            }
+
+            ParserUtils.GetMoreBytesFromStream(this.stream, ref this.buffer, ref reader);
+
+            this.isFinalBlock = reader.IsFinalBlock;
+            this.readerState = reader.CurrentState;
+
+            return result;
+        }
+        catch (JsonException ex)
+        {
+            throw new ParserException($"Error parsing json at position {this.stream.Position}", ex);
+        }
+    }
+
+    private object GetObject(Type type, ref Utf8JsonReader reader)
+    {
+        var jsonObject = ParserUtils.ParseObject(this.stream, ref this.buffer, ref reader);
+        object? result = jsonObject;
+        if (type != typeof(JsonNode))
+        {
+            result = jsonObject.Deserialize(type, this.jsonSerializerOptions);
+        }
+
+        if (result is null)
+        {
+            throw new NotImplementedException();
```

Maybe throw InvalidOperationException here instead?